integrations/groq/src/index.ts

```diff
@@ -0,0 +1,20 @@
+import OpenAI from 'openai'
+import { GenerateContentInput, openai } from '@botpress/common'
+import * as bp from '.botpress'
+
+const groqClient = new OpenAI({
+  baseURL: 'https://api.groq.com/openai/v1',
+  apiKey: bp.secrets.GROQ_API_KEY,
+})
+
+export default new bp.Integration({
+  register: async () => {},
+  unregister: async () => {},
+  actions: {
+    generateContent: async ({ input, logger }) => {
+      return await openai.generateContent(<GenerateContentInput>input, groqClient, logger, { provider: 'groq' })
```

Comment:
**question:** why is casting needed here ?

Reply:
Because zod infer makes properties with `.default()` non-undefined while the action types built for the integrations do so there's a mismatch trying to pass an object with some fields allowing `undefined` in their union type (the action input) in a parameter that doesn't allow them (the `GenerateContentInput` type from zod infer). Not sure why zod infer doesn't specify `undefined` as a possible union type value when using `.default()` but that's basically the issue. If you know a better workaround for this please let me know 🙂